Tests/XCUITests/HistoryTests.swift

```diff
@@ -103,215 +98,214 @@ class HistoryTests: BaseTestCase {
+        // Toggle to private mode. The closed tab is listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
         navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
-        navigator.performAction(Action.OpenNewTabFromTabTray)
-        navigator.nowAt(NewTabScreen)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
         navigator.goto(HistoryRecentlyClosed)
         waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
     }
 
-    func testClearRecentlyClosedHistory() {
-        // Open the default website
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        navigator.nowAt(BrowserTab)
+    func testRemoveAllTabsButton() {
+        openBookOfMozilla()
+
+        // Tap "Remove All Tabs" instead of close the tab individually
         waitForTabsButton()
         navigator.goto(TabTray)
         navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
-        navigator.goto(HistoryRecentlyClosed)
-        // Once the website is visited and closed it will appear in Recently Closed Tabs list
-        waitForExistence(app.tables["Recently Closed Tabs List"])
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
 
-        navigator.goto(HomePanelsScreen)
-        navigator.nowAt(NewTabScreen)
+        // The closed tab is *not* listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
+        navigator.goto(LibraryPanel_History)
+        waitForExistence(app.tables[HistoryPanelA11y.tableView])
+        XCTAssertTrue(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
+        XCTAssertFalse(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+    }
+
+    func testClearRecentlyClosedHistory() {
+        openBookOfMozilla()
 
-        // Go to settings and clear private data
+        closeFirstTabByX()
+
+        // Clear all private data via the settings
+        navigator.nowAt(HomePanelsScreen)
         navigator.goto(ClearPrivateDataSettings)
         app.tables.cells["ClearPrivateData"].tap()
         app.alerts.buttons["OK"].tap()
 
-        // Back on History panel view check that there is not any item
-        navigator.goto(HomePanelsScreen)
-        waitForExistence(app.buttons["urlBar-cancel"], timeout: 5)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(HistoryRecentlyClosed)
-        waitForNoExistence(app.tables["Recently Closed Tabs List"])
+        // The closed tab is *not* listed in "Recently Closed Tabs List"
+        navigator.goto(LibraryPanel_History)
```

Comment:
After tapping "OK" button you are in that menu inside Settings, from here going to LibraryPanel_History implies passing the HomeScreen, and that screen is shown with the keyboard, you need to close it again. Same solution as explained above...in this case before that line you should add:
        ```
navigator.goto(HomePanelsScreen)
        waitForExistence(app.buttons["urlBar-cancel"], timeout: 5)
        navigator.performAction(Action.CloseURLBarOpen)
        navigator.nowAt(NewTabScreen)
```